File: apps/mobile/src/services/api/songs.ts
import { api } from '@services/api';

export interface Song {
  key: string;
  title_ko: string;
  title_en: string;
  composer: string;
  duration_seconds: number;
}

export interface SongListResponse {
  songs: Song[];
}

export interface PreviewUrlResponse {
  song_key: string;
  preview_url: string;
  expires_in_seconds: number;
}

export const songsApi = {
  listSongs: (): Promise<SongListResponse> =>
    api.get('/songs').then(r => r.data),

  getPreviewUrl: (songKey: string): Promise<PreviewUrlResponse> =>
    api.get(`/songs/${songKey}/preview`).then(r => r.data),
};

File: apps/api/app/schemas/songs.py
from pydantic import BaseModel


class SongResponse(BaseModel):
    key: str
    title_ko: str
    title_en: str
    composer: str
    duration_seconds: int


class SongListResponse(BaseModel):
    songs: list[SongResponse]


class PreviewUrlResponse(BaseModel):
    song_key: str
    preview_url: str  # presigned URL (만료 S3_PREVIEW_EXPIRY_SECONDS)
    expires_in_seconds: int
